addons/mail/static/src/js/document_viewer.js

```diff
@@ -0,0 +1,285 @@
+            'click .o_zoom_out': '_onZoomOut',
+            'click .o_close_btn, .o_viewer_img_wrapper': '_onClose',
+            'click .o_print_btn': '_onPrint',
+            'DOMMouseScroll .o_viewer_content': '_onScroll',    // Firefox
+            'mousewheel .o_viewer_content': '_onScroll',        // Chrome, Safari, IE
+            'keydown': '_onKeydown',
+            'mousedown .o_viewer_img': '_startDrag',
+            'mousemove .o_viewer_content': '_doDrag',
+            'mouseup .o_viewer_content': '_endDrag'
+        },
+        /**
+         * When initialize document viewer get image and video type attachment from thread
+         * and store it in instance so we can traversal attachment on next previous
+         *
+         * keep track of currently loaded attachment by 'active_attachment'
+         *
+         * @param {OdooWidget} thread
+         */
+        init: function (thread) {
+            var attachment_ids = thread.get('attachment_ids');
+            this.active_attachment = false;
+            this.scale = 1;
+            this.attachment_ids = _.filter(attachment_ids, function(attachment) {
+                var type = attachment.mimetype.split('/').shift();
+                if (type === 'image' || type === 'video') {
+                    attachment.type = type;
+                    return true;
+                }
+            });
+        },
+        //--------------------------------------------------------------------------
+        // Public
+        //--------------------------------------------------------------------------
+
+        /**
+         * renderContent call after set 'active_attachment' to reflect change on DOM.
+         * used after next, previous to display new content(image/video)
+         */
+        renderContent: function () {
+            this.$('.o_viewer_content').html(QWeb.render('DocumentViewer.Content', {
+                widget: this
+            }));
+            this.$('.o_viewer_img').load(_.bind(this._onImageLoaded, this));
+            this._reset();
+        },
+
+        /**
+         * Open popup/modal with given attachment_id
+         * @param  {integer} attachment_id
+         */
+        on_attachment_popup: function (attachment_id) {
+            this.active_attachment = _.findWhere(this.attachment_ids, {
+                id: attachment_id
+            });
+            this.renderElement();
+            this.$el.modal('show');
+            this.$el.on('hidden.bs.modal', _.bind(this._onDestroy, this));
+            this.$('.o_viewer_img').load(_.bind(this._onImageLoaded, this));
+            this._reset();
+        },
+
+        //--------------------------------------------------------------------------
+        // Private
+        //---------------------------------------------------------------------------
+
+        /**
+         * Remove loading indicator when image loaded
+         * @private
+         */
+        _onImageLoaded: function (){
```

Comment:
Move all handlers in a 'Handlers' section below the 'Private' one. See [here](https://github.com/odoo/odoo/blob/saas-16/addons/web/static/src/js/chrome/pager.js) for example.